checkUrlExists/lambda/index.js

```diff
@@ -50,10 +40,50 @@ async function checkUrl(url) {
 }
 
 async function run(url) {
-  const curlOpts = {
-    caInfo: certFilePath,
-  };
-  return curly.head(url, curlOpts);
+  const parsedUrl = Url.parse(url);
+  const protocol = parsedUrl.protocol; // Url.parse() lower cases the protocol
+  if (FTP_PROTOCOL === protocol || SECURE_FTP_PROTOCOL === protocol) {
+    const secure = SECURE_FTP_PROTOCOL === protocol;
+    const ftpClient = new ftp.Client();
+    try {
+      let options = {
+        host: parsedUrl.host,
+        secure: secure,
+      };
+      if (parsedUrl.port) {
+        options = { port: parsedUrl.port, ...options };
+      }
```

Comment:
There is a pattern to conditionally add a property:
```
{ ...(condition) && {name: "value"} }
```
Applied here, it results in:
```
let options = {
  host: parsedUrl.host,
  secure: secure,
  ...(parsedUrl.port) && { port: parsedUrl.port }
}
```
Not convinced it's better, but definitely shorter!

Reply:
Interesting, did not know that one.